src/Parser.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Worksome\Number;
+
+use BCMath\Number as BCNumber;
+use InvalidArgumentException;
+
+class Parser
+{
+    /**
+     * Parse the given number and extract the whole number and decimal number
+     *
+     * @return array<int, string>
+     */
+    public static function parseFragments(Number|BCNumber|string|int|float $num): array
```

Comment:
Hmm...

Reply:
Got a better solution? haha 👀  I'll give it some more thought.

Reply:
Refactored to be slightly cleaner (no array unpacking). Now there's two methods: `parseWholeNumber` and `parseDecimalNumber` which can be invoked individually:

Was:

```php
[$whole, $decimal] = Parser::parseFragments($number);
```

Now:

```php
$whole = Parser::parseWholeNumber($number);
$decimal = Parser::parseDecimalNumber($number);
```